tools/onnx_subgraph/src/lib/device.cpp

```diff
@@ -15,11 +15,134 @@
  */
 #include "device.h"
 
-void Device::GetDeviceJson(const std::string &json_path) { return; }
+void Device::GetDeviceJson(const std::string &json_path)
+{
+  Json::Reader reader;
+  Json::Value root;
+
+  // Open the JSON file in binary mode
+  std::ifstream in(json_path, std::ios::binary);
+  if (!in.is_open())
+  {
+    std::cout << "Error opening file\n";
```

Comment:
below code used this
```suggestion
    std::cerr << "Error opening file." << std::endl;
```
maybe use single convention